src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -288,18 +288,24 @@ class MergeTreeData : public IStorage, public WithMutableContext
         {
         }
 
-        void addPart(const String & old_name, const String & new_name);
+        void addPart(const String & old_name, const String & new_name, const DiskPtr & disk);
 
         /// Renames part from old_name to new_name
         void tryRenameAll();
 
         /// Renames all added parts from new_name to old_name if old name is not empty
         ~PartsTemporaryRename();
 
+        struct RenameInfo
+        {
+            String old_name;
+            String new_name;
+            DiskPtr disk;
```

Comment:
As I can guess, both old and new part should be on the same disk?
Maybe add a comment here ...